File: packages/storage/ios/Plugin/Classes/Results/ListFilesResult.swift
import Foundation
import Capacitor
import FirebaseStorage

@objc public class ListFilesResult: NSObject, Result {
    private var listResult: StorageListResult

    init(result: StorageListResult) {
        self.listResult = result
    }

    public func toJSObject() -> AnyObject {
        var itemsResult = JSArray()
        for item in listResult.items {
            var itemResult = JSObject()
            itemResult["bucket"] = item.bucket
            itemResult["path"] = item.fullPath
            itemResult["name"] = item.name
            itemsResult.append(itemResult)
        }

        var result = JSObject()
        result["items"] = itemsResult
        result["nextPageToken"] = listResult.pageToken
        return result as AnyObject
    }
}

File: packages/storage/android/src/main/java/app/capgo/capacitor/firebase/storage/ListFilesResult.java
package app.capgo.capacitor.firebase.storage.classes.results;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.google.firebase.storage.ListResult;
import app.capgo.capacitor.firebase.storage.interfaces.Result;

public class ListFilesResult implements Result {

    private ListResult listResult;

    public ListFilesResult(ListResult listResult) {
        this.listResult = listResult;
    }

    public JSObject toJSObject() {
        JSArray itemsResult = new JSArray();
        for (int i = 0; i < listResult.getItems().size(); i++) {
            JSObject itemResult = new JSObject();
            itemResult.put("bucket", listResult.getItems().get(i).getBucket());
            itemResult.put("path", listResult.getItems().get(i).getPath());
            itemResult.put("name", listResult.getItems().get(i).getName());
            itemsResult.put(itemResult);
        }

        JSObject result = new JSObject();
        result.put("items", itemsResult);
        result.put("nextPageToken", listResult.getPageToken());
        return result;
    }
}
